### PR TITLE
Major overhaul!

### LINKED ISSUE #18: Update react-hot-loader 3 once it leaves beta
_From @satazor on February 12, 2017 21:13_

https://github.com/gaearon/react-hot-loader

_Copied from original issue: moxystudio/react-with-moxy#12_

### PR DESCRIPTION
**Pull request by satazor**
Here's a quick summary:

- Embrace env variables + [dotenv](https://www.npmjs.com/package/dotenv)
- Update to webpack 3 and react 16
- Use our [babel preset](https://github.com/moxystudio/babel-preset-moxy) 
- Update all bin files and move them to scripts, to match the upcoming move of those files to react-with-moxy-scripts
- Update [_reboot.css](https://github.com/twbs/bootstrap/blob/v4-dev/dist/css/bootstrap-reboot.css)
- Add test script that uses [Jest](http://facebook.github.io/jest/)
- Add [commitlint](http://marionebl.github.io/commitlint/#/) and [standard-version](https://github.com/conventional-changelog/standard-version): embrace conventional commits!
- Use [webpack-isomorphic-compiler](https://github.com/moxystudio/webpack-isomorphic-compiler) and [webpack-isomorphic-dev-middleware](https://github.com/moxystudio/webpack-isomorphic-dev-middleware)
- Use the new [uglify-es](https://npmjs.com/package/uglify-es)
- And a lot of minor improvements!

Closes #18, #15, #12, #11, #5, #4.

-----------------------------------------

Here's the bundle size improvements:

**before**:
screen shot 2017-11-20 at 00 26 56

**after:**
screen shot 2017-11-20 at 00 27 11

-----------------------------------------------

What's next:

- Create a postcss-preset-moxy
- Create a proper README

### CODE REVIEW

File: scripts/start [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    app.use((req, res, next) => {
+        res.locals.exports = data.exports;
+        res.locals.buildManifest = data.buildManifest;
+        next();
+    });
+
+    // If it's not a public file, render the app!
+    app.use(
+        // There's no compilation to be done, just set the `res.locals`
+        (req, res, next) => {
```
Comment (andrefgneves) [inline scripts/start:108]: Isn't this block duplicated? Locals are already being copied above

File: scripts/start-dev [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    }));
+
+    // If it's not a public file, render the app!
+    app.use(
+        // Setup compilaton so that changes are compiled on the fly!
+        webpackIsomorphicDevMiddleware(isomorphicCompiler, {
+            memoryFs,
+            watchDelay: 50,
+            watchOptions: { poll },
+            notify: { icon: `${publicDir}/favicon.ico` }, // OS notitifications!
```
Comment (andrefgneves) [inline scripts/start-dev]: `notitifications` -> `notifications`

File: src/App.js [hunk 3]
```diff
@@ -6,6 +6,10 @@ import Footer from './footer/Footer';
 import styles from './App.css';
 import { default as buildPageRoutes } from './pages/buildRoutes';
```
Comment (andrefgneves) [inline src/App.js]:
```js
import buildPageRoutes from './pages/buildRoutes';
```

File: scripts/start (hunk 1 above)
Comment (satazor) [reply scripts/start:108]: Yep it is. removed!

File: scripts/start-dev (hunk 2 above)
Comment (satazor) [reply scripts/start-dev]: Fixed.

File: src/App.js (hunk 3 above)
Comment (satazor) [reply src/App.js]: 😓

**Pull request merged**